components/statusLabel/StatusLabel.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import omit from 'lodash.omit';
+import theme from './theme.css';
+
+class StatusLabel extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    className: PropTypes.string,
+    mint: PropTypes.bool,
+    teal: PropTypes.bool,
+    violet: PropTypes.bool,
+    ruby: PropTypes.bool,
+    gold: PropTypes.bool,
+    aqua: PropTypes.bool,
+    small: PropTypes.bool,
+  };
+
+  getColor () {
+    const colors = [
+      'mint',
+      'violet',
+      'ruby',
+      'gold',
+      'aqua',
+      'teal',
+    ];
+
+    for (var i = 0; i < colors.length; i++) {
+      const color = colors[i];
+      if (this.props[color]) {
+        return color;
+      }
+    }
```

Comment:
Doesn't something like this work?

```js
color = colors.filter(color => this.props.hasOwnProperty(color))
return color[0]
```

Reply:
It's starting to get copy/pasted all over the place too, time to extract? :)

Reply:
I actually don't link the filter, because it can't return "early", it keeps evaluating even after it found a match.
Extracting it probably wouldn't hurt indeed.